zrml/swaps/src/tests.rs

```diff
@@ -51,6 +51,28 @@ const _105: u128 = 105 * BASE;
 const _1234: u128 = 1234 * BASE;
 const _10000: u128 = 10000 * BASE;
 
+#[test_case(vec![ASSET_A, ASSET_A]; "case-1 with duplicate assets")]
+#[test_case(vec![ASSET_A, ASSET_B, ASSET_C, ASSET_D, ASSET_E, ASSET_A]; "case-2 with duplicate assets")]
+#[test_case(vec![ASSET_A, ASSET_B, ASSET_C, ASSET_D, ASSET_E, ASSET_E]; "case-3 with duplicate assets")]
+#[test_case(vec![ASSET_A, ASSET_B, ASSET_C, ASSET_A, ASSET_E, ASSET_D]; "case-4 with duplicate assets")]
+fn pool_creation_fails_with_duplicate_assets(assets: Vec<Asset<<Runtime as Config>::MarketId>>) {
```

Comment:
I usually go by this pattern when naming tests: `{function_under_test}_{behavior_during_test}`

```suggestion
fn create_pool_fails_with_duplicate_assets(assets: Vec<Asset<<Runtime as Config>::MarketId>>) {
```